### PR TITLE
layout: Place inline absolute hypothetical boxes properly during block fragment position assignment.

### LINKED ISSUE #13471: Baselines of inline absolute hypothetical boxes not properly aligned
Test case:

    <!DOCTYPE html>
    <html>
    <meta charset="utf-8">
    <div style="border-bottom: solid black 1px;">A<span style="position: absolute; right: 0;">B</span></div>

This results in the "src" link being placed in the wrong place in Rust documentation. Oddly, sometimes it works and sometimes it doesn't in rustdoc (probably due to incremental reflow as style sheets/images are loaded), but this minimal test case always fails for me.

### PR DESCRIPTION
**Pull request by pcwalton**
r? @notriddle

Closes #13471.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13490)

### CODE REVIEW

**Comment by pcwalton** [comment]
Closes #13471.

File: components/layout/fragment.rs [hunk 1]
```diff
@@ -2036,7 +2036,7 @@ impl Fragment {
     /// Calculates block-size above baseline, depth below baseline, and ascent for this fragment
     /// when used in an inline formatting context. See CSS 2.1 § 10.8.1.
     pub fn inline_metrics(&self, layout_context: &LayoutContext) -> InlineMetrics {
-        match self.specific {
+        return match self.specific {
```
Comment (notriddle) [inline components/layout/fragment.rs]: Why did you add the return statement here?

**Comment by notriddle** [comment]
Also, could you add a spec reference?

**Comment by notriddle** [comment]
```
  • 18 tests failed unexpectedly
  • 31 tests passed unexpectedly
```

**Comment by pcwalton** [comment]
@notriddle: This behavior is unspecified. The spec literally says "a user agent is free to take a guess as to [the hypothetical box]'s probable position".

**Comment by notriddle** [comment]
@pcwalton Is there a spec bug for this?

**Comment by pcwalton** [comment]
@SimonSapin Has anyone brought up the need to specify hypothetical box placement better in the various WGs?

**Comment by notriddle-cerebrum** [comment]
@pcwalton Might be interesting:

* http://searchfox.org/mozilla-central/source/layout/generic/ReflowInput.cpp#1209
* https://trac.webkit.org/browser/trunk/Source/WebCore/rendering/RenderBox.cpp#L3499

**Comment by notriddle** [comment]
@pcwalton Might be interesting:

* http://searchfox.org/mozilla-central/source/layout/generic/ReflowInput.cpp#1209
* https://trac.webkit.org/browser/trunk/Source/WebCore/rendering/RenderBox.cpp#L3499

*Edit: wrong browser window.*

**Comment by pcwalton** [comment]
Rewritten per IRC discussion.

r? @notriddle

**Review by notriddle — APPROVED** [review]
